torch/ao/quantization/fx/backend_config/native.py

```diff
@@ -78,6 +82,31 @@ def _get_default_op_backend_config(op, dtype_configs):
         torch.nn.functional.layer_norm,
     ]]
 
+
+def _binary_op_observation_type_getter(quantize_handler):
+    # determine how many of the first two args are Tensors (versus scalars)
+    # this distinguishes things like "x + y" from "x + 2" or "2 + x"
+    num_tensor_args = 0
+    cache_for_no_tensor_check: Dict[Node, bool] = dict()
+    for arg_idx in range(len(quantize_handler.root_node.args)):
+        arg = quantize_handler.root_node.args[arg_idx]
+        if isinstance(arg, Node) and (not all_node_args_have_no_tensors(arg, quantize_handler.modules, cache_for_no_tensor_check)):
+            num_tensor_args += 1
+    if num_tensor_args == 2:
+        return ObservationType.OUTPUT_USE_DIFFERENT_OBSERVER_AS_INPUT
+    else:
+        return ObservationType.OUTPUT_SHARE_OBSERVER_WITH_INPUT
+
+_ADD_CONFIG = {
+    "pattern": operator.add,
```

Comment:
this operator is overloaded, should we make the configs match the quantized ops (multiple configs, one per overload), instead of forking a single op config to handle the overloads?

Reply:
are you talking about `torch.add`? like https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L442 and https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L501?
I feel in general the overload info is not available to us, from our perspective we only see a floating point add operator in the graph, are you suggesting to infer the overload based on the dtype input of the operator?

Reply:
we have some design that takes types like (Tensor, Scalar) (maybe enum) before, but I'm not aware of an easy way to enable these things in the current codebase, I think this will require type inference, which we don't have currently. I'm open to change the api/implementation after we have the type inference ready in the future

Reply:
ok, how about just enabling different settings based on number of tensor arguments

Reply:
sure, we can do that, would this be to specific to the use case we have? do we plan to change it to specifying types after we have dtype inference?

Reply:
looks like the complexity is coming from here:
1. `torch.add` is a single operator in user code
2. `torch.add` calls different kernels (in fp32 or int8 or other dtypes) depending on # of tensor arguments

Do you want your config design to be unique by the operator type in user code (and have to support forking of each entry), or do you want your config design to be unique by kernel (and have to support one to many map of operator in user code to config entry)?  I think they are both correct, just a matter of which one you think fits better in your design.

Reply:
Yes, I think so. I feel it makes more sense to config the design to be unique by operator type/fusion pattern, since that is what we are matching against, otherwise the matching logic will be complex and will be subject to future extensions, e.g. if we have other variants that depend on other things, the matching logic need to change again to accommodate the changes.
This choice will simplify our implementation and make our matching code much more maintainable and understandable.

Reply:
using "_is_binary_op_with_binary_scalar_op_variant" and hided the internal implementation details, please take a look again